src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -114,18 +114,34 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
       this.trace('startVideoPreviewForVideoInput', element.id);
       return;
     }
-    this.releaseMediaStream(element.srcObject as MediaStream);
-    DefaultVideoTile.disconnectVideoStreamFromVideoElement(element);
-    navigator.mediaDevices
-      .getUserMedia(this.activeDevices['video'].constraints)
-      .then(previewStream => {
-        DefaultVideoTile.connectVideoStreamToVideoElement(previewStream, element, true);
-      })
-      .catch(error => {
-        this.logger.warn(
-          `Unable to reacquire video stream for preview to element ${element.id}: ${error}`
-        );
-      });
+
+    const attachedMediaStream = element.srcObject as MediaStream;
+    if (attachedMediaStream && attachedMediaStream.active) {
+      this.logger.warn('video element for preview still holding reference to active stream');
+      this.releaseMediaStream(attachedMediaStream);
+    }
+    if (this.activeDevices['video'].stream && this.activeDevices['video'].stream.active) {
+      // bind a cloned MediaStream of chosen activeDevices stream
+      // API users are responsible to managing video preview MediaStream. And ensure stopVideoPreviewForVideoInput is called when end the preview
+      DefaultVideoTile.connectVideoStreamToVideoElement(
+        this.activeDevices['video'].stream.clone(),
+        element,
+        true
+      );
+    } else {
+      this.logger.warn('startVideoPreviewForVideoInput before any videoInputDevice is chosen');
```

Comment:
I don't understand how this edge case is different than the one at the top of the function:

```
if (!this.activeDevices['video']) {
  this.logger.warn('cannot bind video preview since video input device has not been chosen');
  ...
```

How would we end up in this scenario given the above guard?

Reply:
This is more of a demo issue, when we choose synthesized device, we have `this.activeDevices['video'].stream` corresponds to a canvas captured stream. Before this PR, `getUserMedia` is always called here and return a actual video stream instead of canvas-captured stream.